keel-core/src/main/kotlin/com/netflix/spinnaker/keel/validators/DeliveryConfigValidator.kt

```diff
@@ -0,0 +1,84 @@
+package com.netflix.spinnaker.keel.validators
+
+import com.netflix.spinnaker.keel.core.api.DependsOnConstraint
+import com.netflix.spinnaker.keel.core.api.SubmittedDeliveryConfig
+import com.netflix.spinnaker.keel.exceptions.DuplicateArtifactReferenceException
+import com.netflix.spinnaker.keel.exceptions.DuplicateResourceIdException
+import com.netflix.spinnaker.keel.exceptions.MissingEnvironmentReferenceException
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+/**
+ * Run validation checks against delivery config to ensure:
+ *
+ * - resources have unique ids
+ * - artifacts have unique references
+ * - depends on environments are unique
+ *
+ * Throws an exception if config fails any checks
+ */
+@Component
+class DeliveryConfigValidator {
+
+  val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  fun validate(config: SubmittedDeliveryConfig) {
```

Comment:
Changed the object to be `SubmittedDeliveryConfig`, all the logic remained the same.